src/Bicep.Core/TypeSystem/Az/AzResourceTypeProvider.cs

```diff
@@ -191,7 +190,7 @@ public static IEnumerable<TypeProperty> KnownTopLevelResourceProperties()
             }, null));
         }
 
-        public AzResourceTypeProvider(IResourceTypeLoader resourceTypeLoader, string providerVersion)
+        public AzResourceTypeProvider(IResourceTypeLoader resourceTypeLoader, string providerVersion = IResourceTypeProvider.BuiltInVersion)
```

Comment:
Optional parameters are often best avoided - forcing the caller to pass this value makes it more explicit, and makes it difficult for new usage to forget to pass the value.

Is there a strong need to set a default value here?